commit a37ddddd86037c896c702b4df416bc4e51b2a5a0
Author: Russ Weight <russell.h.weight@intel.com>
Date:   Tue Apr 26 09:35:32 2022 -0700

    selftests: firmware: Add firmware upload selftests
    
    Add selftests to verify the firmware upload mechanism. These test
    include simple firmware uploads as well as upload cancellation and
    error injection. The test creates three firmware devices and verifies
    that they all work correctly and independently.
    
    Tested-by: Matthew Gerlach <matthew.gerlach@linux.intel.com>
    Reviewed-by: Luis Chamberlain <mcgrof@kernel.org>
    Reviewed-by: Tianfei zhang <tianfei.zhang@intel.com>
    Signed-off-by: Russ Weight <russell.h.weight@intel.com>
    Link: https://lore.kernel.org/r/20220426163532.114961-1-russell.h.weight@intel.com
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/tools/testing/selftests/firmware/Makefile b/tools/testing/selftests/firmware/Makefile
index 40211cd8f0e6..7992969deaa2 100644
--- a/tools/testing/selftests/firmware/Makefile
+++ b/tools/testing/selftests/firmware/Makefile
@@ -4,7 +4,7 @@ CFLAGS = -Wall \
          -O2
 
 TEST_PROGS := fw_run_tests.sh
-TEST_FILES := fw_fallback.sh fw_filesystem.sh fw_lib.sh
+TEST_FILES := fw_fallback.sh fw_filesystem.sh fw_upload.sh fw_lib.sh
 TEST_GEN_FILES := fw_namespace
 
 include ../lib.mk